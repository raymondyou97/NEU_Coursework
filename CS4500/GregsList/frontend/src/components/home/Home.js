import React from 'react';

import Button from 'react-bootstrap/Button';
import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';
import ButtonGroup from 'react-bootstrap/ButtonGroup';

import '../../styles/home.css';

class Home extends React.PureComponent {
  componentDidMount() {
    this.props.fetchServiceCategories();
    this.props.fetchTabServiceCategories(6);
    this.props.fetchLoggedInUser();
  }

  handleTermChange = e => {
    this.props.handleTermChange(e.target.value);
  };

  handleZipChange = e => {
    this.props.handleZipChange(e.target.value);
  };

  renderHeader() {
    const { loggedInUser, logOut } = this.props;

    const links = loggedInUser ? (
      <div className="header-links">
        <span className="user-name">Welcome {loggedInUser}!</span>
        <span className="log-out" onClick={logOut}>
          Log out
        </span>
      </div>
    ) : (
      <div className="header-links">
        <a className="link" href="SignUp">
          Sign up
        </a>
        <a className="link" href="LogIn">
          Log in
        </a>
      </div>
    );

    return (
      <div className="header">
        <h1>Find professionals near you</h1>
        {links}
      </div>
    );
  }

  renderSearchSection() {
    return (
      <div className="search-section">
        <InputGroup className="search-bars">
          <FormControl
            placeholder="Search for providers"
            onChange={this.handleTermChange}
          />
          <FormControl placeholder="Zip code" onChange={this.handleZipChange} />
          <InputGroup.Append>
            <Button
              className="search-button"
              variant="primary"
              onClick={() => this.props.handleSearch()}
            >
              Search
            </Button>
          </InputGroup.Append>
        </InputGroup>
      </div>
    );
  }

  renderServiceButtons() {
    const { serviceCategories } = this.props;

    if (!(serviceCategories && serviceCategories.length)) return null;

    return (
      <div className="service-categories-container">
        <ButtonGroup className="service-categories">
          {serviceCategories[0] ? (
            <Button variant="link">
              <a href={`services#${serviceCategories[0].id}`}>
                <div className="service-category-button">
                  {serviceCategories[0].title}
                </div>
              </a>
            </Button>
          ) : null}
          {serviceCategories[1] ? (
            <Button variant="link">
              <a href={`services#${serviceCategories[1].id}`}>
                <div className="service-category-button">
                  {serviceCategories[1].title}
                </div>
              </a>
            </Button>
          ) : null}
          {serviceCategories[2] ? (
            <Button variant="link">
              <a href={`services#${serviceCategories[2].id}`}>
                <div className="service-category-button">
                  {serviceCategories[2].title}
                </div>
              </a>
            </Button>
          ) : null}
          {serviceCategories[3] ? (
            <Button variant="link">
              <a href={`services#${serviceCategories[3].id}`}>
                <div className="service-category-button">
                  {serviceCategories[3].title}
                </div>
              </a>
            </Button>
          ) : null}
          {serviceCategories.length > 4 ? (
            <Button variant="link">
              <a href="services">
                <div className="service-category-button">More</div>
              </a>
            </Button>
          ) : null}
        </ButtonGroup>
      </div>
    );
  }

  generateSelectedCategoryTab() {
    if (this.props.selectedTab !== -1) {
      return (
        <div className="row">
          {this.props.tabServiceCategories[this.props.selectedTab].services.map(
            (service, index) => (
              <div key={service.id} className="card col-4 no-border">
                <img
                  src="/images/skywhale.jpg"
                  className="card-img-top"
                  alt="..."
                />
                <div className="card-body">
                  <h5 className="card-title">
                    <a href={`/ServiceProviderSearch/${service.id}`}>
                      {service.title}
                    </a>
                  </h5>
                </div>
              </div>
            )
          )}
        </div>
      );
    }
  }

  renderServiceTabs() {
    const { tabServiceCategories, selectServiceTab } = this.props;

    if (!tabServiceCategories.length) return null;

    return (
      <div>
        <ul className="nav nav-tabs">
          {tabServiceCategories.map((serviceCategory, index) => (
            <li key={serviceCategory.id} className="nav-item">
              {/* eslint-disable-next-line */}
              <a
                onClick={() => selectServiceTab(index)}
                className="nav-link"
                href="#"
              >
                {serviceCategory.title}
              </a>
            </li>
          ))}
        </ul>
        {this.generateSelectedCategoryTab()}
        <br />
      </div>
    );
  }

  render() {
    return (
      <div className="home">
        {this.renderHeader()}
        {this.renderSearchSection()}
        {this.renderServiceButtons()}
        {this.renderServiceTabs()}
      </div>
    );
  }
}

export default Home;
