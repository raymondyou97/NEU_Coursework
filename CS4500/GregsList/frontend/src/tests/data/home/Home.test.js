'use es6';

import React from 'react';
import Home from '../../../components/home/Home';
import renderer from 'react-test-renderer';
import { ServiceCategories, ThreeServiceCategories } from './HomeComponentData';

test('Empty snapshot renders', () => {
  const component = renderer.create(
    <Home
      fetchServiceCategories={() => {}}
      fetchTabServiceCategories={() => {}}
      fetchLoggedInUser={() => {}}
      serviceCategories={[]}
      tabServiceCategories={[]}
      selectServiceTabs={() => {}}
      selectedTab={-1}
      loggedInUser={null}
      logOut={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot with data renders', () => {
  const component = renderer.create(
    <Home
      fetchServiceCategories={() => {}}
      fetchTabServiceCategories={() => {}}
      fetchLoggedInUser={() => {}}
      serviceCategories={ServiceCategories}
      tabServiceCategories={ServiceCategories}
      selectServiceTabs={() => {}}
      selectedTab={-1}
      loggedInUser={null}
      logOut={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot with less data renders', () => {
  const component = renderer.create(
    <Home
      fetchServiceCategories={() => {}}
      fetchTabServiceCategories={() => {}}
      fetchLoggedInUser={() => {}}
      serviceCategories={ThreeServiceCategories}
      tabServiceCategories={ThreeServiceCategories}
      selectServiceTabs={() => {}}
      selectedTab={-1}
      loggedInUser={null}
      logOut={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot with tab selected renders', () => {
  const component = renderer.create(
    <Home
      fetchServiceCategories={() => {}}
      fetchTabServiceCategories={() => {}}
      fetchLoggedInUser={() => {}}
      serviceCategories={ServiceCategories}
      tabServiceCategories={ServiceCategories}
      selectServiceTabs={() => {}}
      selectedTab={0}
      loggedInUser={null}
      logOut={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
