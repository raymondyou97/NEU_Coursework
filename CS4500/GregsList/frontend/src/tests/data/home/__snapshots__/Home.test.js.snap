// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty snapshot renders 1`] = `
<div
  className="home"
>
  <div
    className="header"
  >
    <h1>
      Find professionals near you
    </h1>
    <div
      className="header-links"
    >
      <a
        className="link"
        href="SignUp"
      >
        Sign up
      </a>
      <a
        className="link"
        href="LogIn"
      >
        Log in
      </a>
    </div>
  </div>
  <div
    className="search-section"
  >
    <div
      className="search-bars input-group"
    >
      <input
        className="form-control"
        placeholder="Search for providers"
      />
      <input
        className="form-control"
        placeholder="Zip code"
      />
      <div
        className="input-group-append"
      >
        <button
          className="search-button btn btn-primary"
          disabled={false}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot with data renders 1`] = `
<div
  className="home"
>
  <div
    className="header"
  >
    <h1>
      Find professionals near you
    </h1>
    <div
      className="header-links"
    >
      <a
        className="link"
        href="SignUp"
      >
        Sign up
      </a>
      <a
        className="link"
        href="LogIn"
      >
        Log in
      </a>
    </div>
  </div>
  <div
    className="search-section"
  >
    <div
      className="search-bars input-group"
    >
      <input
        className="form-control"
        placeholder="Search for providers"
      />
      <input
        className="form-control"
        placeholder="Zip code"
      />
      <div
        className="input-group-append"
      >
        <button
          className="search-button btn btn-primary"
          disabled={false}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </div>
  <div
    className="service-categories-container"
  >
    <div
      className="service-categories btn-group"
      role="group"
    >
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services#1"
        >
          <div
            className="service-category-button"
          >
            test1
          </div>
        </a>
      </button>
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services#2"
        >
          <div
            className="service-category-button"
          >
            test2
          </div>
        </a>
      </button>
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services#3"
        >
          <div
            className="service-category-button"
          >
            test3
          </div>
        </a>
      </button>
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services#4"
        >
          <div
            className="service-category-button"
          >
            test4
          </div>
        </a>
      </button>
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services"
        >
          <div
            className="service-category-button"
          >
            More
          </div>
        </a>
      </button>
    </div>
  </div>
  <div>
    <ul
      className="nav nav-tabs"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test1
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test2
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test3
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test4
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test5
        </a>
      </li>
    </ul>
    ;
    <br />
  </div>
</div>
`;

exports[`Snapshot with less data renders 1`] = `
<div
  className="home"
>
  <div
    className="header"
  >
    <h1>
      Find professionals near you
    </h1>
    <div
      className="header-links"
    >
      <a
        className="link"
        href="SignUp"
      >
        Sign up
      </a>
      <a
        className="link"
        href="LogIn"
      >
        Log in
      </a>
    </div>
  </div>
  <div
    className="search-section"
  >
    <div
      className="search-bars input-group"
    >
      <input
        className="form-control"
        placeholder="Search for providers"
      />
      <input
        className="form-control"
        placeholder="Zip code"
      />
      <div
        className="input-group-append"
      >
        <button
          className="search-button btn btn-primary"
          disabled={false}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </div>
  <div
    className="service-categories-container"
  >
    <div
      className="service-categories btn-group"
      role="group"
    >
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services#1"
        >
          <div
            className="service-category-button"
          >
            test1
          </div>
        </a>
      </button>
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services#2"
        >
          <div
            className="service-category-button"
          >
            test2
          </div>
        </a>
      </button>
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services#3"
        >
          <div
            className="service-category-button"
          >
            test3
          </div>
        </a>
      </button>
    </div>
  </div>
  <div>
    <ul
      className="nav nav-tabs"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test1
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test2
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test3
        </a>
      </li>
    </ul>
    ;
    <br />
  </div>
</div>
`;

exports[`Snapshot with tab selected renders 1`] = `
<div
  className="home"
>
  <div
    className="header"
  >
    <h1>
      Find professionals near you
    </h1>
    <div
      className="header-links"
    >
      <a
        className="link"
        href="SignUp"
      >
        Sign up
      </a>
      <a
        className="link"
        href="LogIn"
      >
        Log in
      </a>
    </div>
  </div>
  <div
    className="search-section"
  >
    <div
      className="search-bars input-group"
    >
      <input
        className="form-control"
        placeholder="Search for providers"
      />
      <input
        className="form-control"
        placeholder="Zip code"
      />
      <div
        className="input-group-append"
      >
        <button
          className="search-button btn btn-primary"
          disabled={false}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </div>
  <div
    className="service-categories-container"
  >
    <div
      className="service-categories btn-group"
      role="group"
    >
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services#1"
        >
          <div
            className="service-category-button"
          >
            test1
          </div>
        </a>
      </button>
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services#2"
        >
          <div
            className="service-category-button"
          >
            test2
          </div>
        </a>
      </button>
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services#3"
        >
          <div
            className="service-category-button"
          >
            test3
          </div>
        </a>
      </button>
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services#4"
        >
          <div
            className="service-category-button"
          >
            test4
          </div>
        </a>
      </button>
      <button
        className="btn btn-link"
        disabled={false}
        type="button"
      >
        <a
          href="services"
        >
          <div
            className="service-category-button"
          >
            More
          </div>
        </a>
      </button>
    </div>
  </div>
  <div>
    <ul
      className="nav nav-tabs"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test1
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test2
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test3
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test4
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="#"
          onClick={[Function]}
        >
          test5
        </a>
      </li>
    </ul>
    <div
      className="row"
    >
      <div
        className="card col-4 no-border"
      >
        <img
          alt="..."
          className="card-img-top"
          src="/images/skywhale.jpg"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            <a
              href="/providers"
            >
              Pet Washing
            </a>
          </h5>
        </div>
      </div>
      <div
        className="card col-4 no-border"
      >
        <img
          alt="..."
          className="card-img-top"
          src="/images/skywhale.jpg"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            <a
              href="/providers"
            >
              Pet Nail Services
            </a>
          </h5>
        </div>
      </div>
      <div
        className="card col-4 no-border"
      >
        <img
          alt="..."
          className="card-img-top"
          src="/images/skywhale.jpg"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            <a
              href="/providers"
            >
              Pet Boarding
            </a>
          </h5>
        </div>
      </div>
      <div
        className="card col-4 no-border"
      >
        <img
          alt="..."
          className="card-img-top"
          src="/images/skywhale.jpg"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            <a
              href="/providers"
            >
              Pet Day Care
            </a>
          </h5>
        </div>
      </div>
      <div
        className="card col-4 no-border"
      >
        <img
          alt="..."
          className="card-img-top"
          src="/images/skywhale.jpg"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            <a
              href="/providers"
            >
              Pet Training
            </a>
          </h5>
        </div>
      </div>
      <div
        className="card col-4 no-border"
      >
        <img
          alt="..."
          className="card-img-top"
          src="/images/skywhale.jpg"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            <a
              href="/providers"
            >
              Pet Home Medical
            </a>
          </h5>
        </div>
      </div>
      <div
        className="card col-4 no-border"
      >
        <img
          alt="..."
          className="card-img-top"
          src="/images/skywhale.jpg"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            <a
              href="/providers"
            >
              Pet Walking
            </a>
          </h5>
        </div>
      </div>
    </div>
    ;
    <br />
  </div>
</div>
`;
